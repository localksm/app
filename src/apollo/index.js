import {
  ApolloClient,
  ApolloLink,
  HttpLink,
  InMemoryCache,
} from 'apollo-boost';
import { WebSocketLink } from 'apollo-link-ws';
import { getMainDefinition } from 'apollo-utilities';
import { onError } from 'apollo-link-error';
import { setContext } from 'apollo-link-context';
import { split } from 'apollo-link';
import ls from 'react-native-local-storage';
import {
  GRAPHQL_ENDPOINT,
  GRAPHQL_SUBSCRIPTIONS_ENDPOINT,
} from '../utils/config';
import { ContextProvider, ContextConsumer, withContext } from './context';
import ApolloState from './stateManager';
import {
  SIGNUP,
  LOGIN,
  ADD_FUNDS,
  CONFIRM_PROPOSAL,
  SEND_ACCEPTANCE,
  SEND_RESOLUTION,
  SEND_SETTLEMENT,
  SEND_FULFILLMENT,
  INSERT_PROPOSAL_PAYMENT_METHOD,
  SEND_DISBURSEMENT_BUYER,
  SEND_DISBURSEMENT_SELLER,
  SELL,
} from './mutations';
import {
  VERIFY_USER,
  FEE,
  PAYMENT_METHODS,
  CURRENCIES,
  COUNTRIES,
  PUBLIC_KEY,
  QUERY_PROPOSALS,
  QUERY_USER_PROPOSALS,
} from './queries';

const httpLink = new HttpLink({
  uri: `${GRAPHQL_ENDPOINT}`,
});

const wsLink = new WebSocketLink({
  uri: `${GRAPHQL_SUBSCRIPTIONS_ENDPOINT}`,
  options: { reconnect: true },
});

export const setSession = async payload => {
  return await ls.save('session', payload);
};

export const getSession = async () => {
  return await ls.get('session');
};

export const removeSession = async () => {
  return await ls.clear();
};

const authLink = setContext(async _ => {
  const data = await getSession();
  const { session } = data !== null && data;
  const jwt = session && session.token;

  return {
    headers: {
      authorization: `Bearer ${jwt}`,
    },
  };
});

const httpAuthLink = authLink.concat(httpLink);

const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) =>
      // eslint-disable-next-line no-console
      console.log(
        `[GraphQL error]: Message: ${message}, Location: ${JSON.stringify(locations)}, Path: ${path}`,
      ),
    );
  }
  // eslint-disable-next-line no-console
  if (networkError) console.log(`[Network error]: ${networkError}`);
});

const link = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === 'OperationDefinition' &&
      definition.operation === 'subscription'
    );
  },
  wsLink,
  httpAuthLink,
);
export const cache = new InMemoryCache();
export const client = new ApolloClient({
  link: ApolloLink.from([errorLink, link]),
  cache,
  resolvers: {
    logout: async id => {
      await removeSession();
    },
  },
});

export const QUERIES = {
  VERIFY_USER,
  FEE,
  PAYMENT_METHODS,
  COUNTRIES,
  CURRENCIES,
  PUBLIC_KEY,
  QUERY_PROPOSALS,
  QUERY_USER_PROPOSALS,
};

export const MUTATIONS = {
  SIGNUP,
  LOGIN,
  SEND_ACCEPTANCE,
  SEND_RESOLUTION,
  SEND_SETTLEMENT,
  SEND_FULFILLMENT,
  INSERT_PROPOSAL_PAYMENT_METHOD,
  ADD_FUNDS,
  CONFIRM_PROPOSAL,
  SEND_DISBURSEMENT_BUYER,
  SEND_DISBURSEMENT_SELLER,
  SELL,
};

export const SUBSCRIPTIONS = {};

export { ContextConsumer, ContextProvider, withContext };

const initialState = {
  session: null,
  offer: null,
  filterData: [],
  navigation: null,
  evidenceImages: null,
  token: null,
};
cache.writeData({ data: initialState });

export const state = ApolloState;
