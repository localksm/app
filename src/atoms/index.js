export { default as Button } from './Button';
export { default as InputText } from './InputText';
export { default as Link } from './Link';
export { default as Dropdown } from './Dropdown';
export { default as FBLoginButton } from './facebookButton';
export { default as GoogleButton } from './GoogleButton';
export { default as ButtonSignUp } from './ButtonSignUp';
export { default as Fees } from './Fees';
export { default as ButtonSignIn } from './ButtonSignIn';
export { default as DropdownPaymentMethods } from './DropdownPaymentMethods';
export { default as DropdownCountries } from './DropdownCountries';
export { default as DropdownCurrencies } from './DropdownCurrencies';
export { default as AddFundsButton } from './AddFundsButton';
export { default as QrWallet } from './QrCode';
export { default as ProposalTypeImages } from './ProposalTypeImages'; 
export { default as ConfirmSentBuyButton } from './ConfirmSentBuyButton';
export { default as ConfirmReceivedButton } from './ConfirmReceivedButton';
export { default as TWLoginButton } from './TwitterButton';
export { default as SellButton } from './SellButton';
export { default as InputLayout} from './InputLayout';