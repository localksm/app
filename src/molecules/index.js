export { default as HeaderLayout } from './HeaderLayout';
export { default as Offer } from './Offer';
export { default as FooterWhite } from './FooterWhite';
export { default as MultiView } from './MultiView';
export { default as CompletedMultiView } from './CompletedMultiView';
export { default as FilePickerComponent } from './FilePickerComponent';
export { default as FilePicker } from './FilePicker';
export { default as WalletContent } from './WalletContent';
export { default as CardProposal } from './CardProposal';
export { default as Card } from './Card';
export { default as BalanceHeader } from './BalanceHeader';
export { default as PaymentForm } from './PaymentForm';
export { default as PaymentFormInputs } from './PaymentFormInputs';
export { default as FormInput } from './FromInput';
export { default as FormTextInp } from './FormTextInp';
export { default as FormTextAreaInp } from './FormTextAreaInp';
export { default as GenericGQLLoading } from './GenericGQLLoading';
export { default as GenericGQLError } from './GenericGQLError';
export { default as ConfirmedBuySellButton } from './ConfirmedBuySellButton';
export { default as WithdrawText } from './WithdrawText';
export { default as WalletSwitch} from './WalletSwitch';
export { default as WalletViewButton } from './WalletViewButton';