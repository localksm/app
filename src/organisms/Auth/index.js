export { default as FormSignIn } from './FormSignIn';
export { default as FormSignUp } from './FormSignUp';
