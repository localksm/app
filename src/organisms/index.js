export { default as HomeLayout } from './HomeLayout';
export { default as OfferList } from './OfferList';
export { default as OfferDetails } from './OfferDetails';
export { default as OfferDetailContent } from './OfferDetailContent';
export { default as FormSignIn } from './FormSignIn';
export { default as FormSignInContent } from './FormSignInContent';
export { default as FormSignUp } from './FormSignUp';
export { default as FormCreateOfferBuy } from './FormCreateOfferBuy';
export { default as FormCreateOfferSell } from './FormCreateOfferSell';
export { default as ConfirmationBuySell } from './ConfirmationBuySell';
export { default as ConfirmedBuySell } from './ConfirmedBuySell';
export { default as AcceptedBuySell } from './AcceptedBuySell';
export { default as DisbursementBuyer } from './DisbursementBuyer';
export { default as DisbursementSeller } from './DisbursementSeller';
export { default as Completed } from './Completed';
export { default as WalletView } from './WalletView';
export { default as Report } from './Report';
export { default as ReportContent } from './ReportContent';
export { default as MediationView } from './MediationView';
export { default as MediationResult } from './MediationResult';
export { default as MyOffer } from './MyOffer';
export { default as Slider } from './Slider';
export { default as OnboardingImage } from './OnboardingImage';
export { default as HelpView } from './HelpView';
export { default as FormLayout } from './FormLayout';
export { default as Withdraw } from './Withdraw';
export { default as FormCreatePin } from './FormCreatePin';
export { default as EnterPin } from './EnterPin';
export { default as SendSettlementMakerContent } from './SendSettlementMakerContent';

export { default as CustomDrawerContent } from './CustomDrawer';
