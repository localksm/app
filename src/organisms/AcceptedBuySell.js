import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MultiView, FooterWhite } from '../molecules';
import { Link, Button } from '../atoms';
import { demo } from '../utils/demoQuery';


const AcceptedBuySell = props => {
  const title = `Please wait until ${demo.body.usernameMaker} sends you`;
  const amountKSM = '00.30 KSM'
  const exchange = `$ ${demo.body.offerAmount} in USD \n ${amountKSM} Local Currency`;
  const details = `Payment method \n ${demo.body.paymentMethod} \nPayment details\n ${demo.body.paymentData.name}\n ${demo.body.paymentData.lastName} \n ${demo.body.paymentData.country} \n ${demo.body.paymentData.email} \n ${demo.body.paymentData.address}`;
  return (
    <View>
      <MultiView
        title={title}
        exchange={exchange}
        details={details}
        stylect={styles.container}>
       {/* the button only appears in status confirmed for the Maker */}
          {props.route.params.type === 'Sell' &&
              <Button label="Confirm sent" />
           
          }
      </MultiView>
      <FooterWhite stylectContainer={styles.buttons}>
          <Link label="Report a problem" color="#cc5741" action={() => props.navigation.navigate('ReportAProblem')} />
          <Link label="Continue Disburse" color="#cc5741" action={() => props.navigation.navigate('Disburse')} />
      </FooterWhite>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingTop: '15%',
  },
  text: {
    fontSize: 14,
    fontFamily: 'Poppins-SemiBold',
  },
  buttons: {
    paddingTop: '1%',
    alignItems: 'center',
    
  },
});

export default AcceptedBuySell;
