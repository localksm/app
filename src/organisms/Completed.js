import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { CompletedMultiView } from '../molecules';
import { Link, Button } from '../atoms';
import { demo } from '../utils/demoQuery';

const Completed = props => {

const title = `Please wait until ${demo.body.usernameMaker} sends you`;
  const amountKSM = '00.30 KSM';
  const exchange = `$ ${
    demo.body.offerAmount
  } in USD \n ${amountKSM} Local Currency`;
  const details = `Payment method \n ${
    demo.body.paymentMethod
  } \nPayment details\n ${demo.body.paymentData.name}\n ${
    demo.body.paymentData.lastName
  } \n ${demo.body.paymentData.country} \n ${demo.body.paymentData.email} \n ${
    demo.body.paymentData.address
  }`;
  return (
    <View>
      <CompletedMultiView
        title={title}
        exchange={exchange}
        details={details}
        stylect={styles.container}>
        <View style={styles.textContainer}>
          <Text style={styles.text}>
              
            {/* here the information of the FEE will be placed */}
          </Text>
        </View>
        <Button label="Ok" action={() => props.navigation.navigate('App')} />

        
       
      </CompletedMultiView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingTop: '15%',
  },
  textContainer: {
    alignItems: 'center',
    paddingBottom: '5%',
  },
  text: {
    fontSize: 14,
    fontFamily: 'Poppins-SemiBold',
  },
});

export default Completed;