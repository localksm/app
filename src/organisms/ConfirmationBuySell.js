import React from 'react';
import { View, Text, StyleSheet, Image } from 'react-native';
import { Link } from '../atoms';
import { FooterWhite } from '../molecules';

const ConfirmationBuySell = props => {
  console.log(props.route.params.variables);
  const {currency, offered, required} = props.route.params.variables
  
  return (
    <View>
      <View style={styles.container}>
        <View>
          <Image source={require('../../assets/submitted.png')} />
        </View>
        <View>
          <Text style={styles.textTitle}>
            {props.route.params.typeOffer} submitted{' '}
          </Text>
        </View>
        <View>
          <Text style={styles.text}>Please wait until someone accepts it.</Text>
        </View>
      </View>

      <FooterWhite stylectContainer={styles.containerColum}>
        <View style={styles.colum}>
          <View>
            <Text style={styles.textColumOneRowOne}>Offered</Text>
          </View>
          <View>
            <Text style={styles.textColumOneRowTwo}>{offered} {currency}</Text>
          </View>
        </View>
        <View style={styles.colum}>
          <View>
            <Text style={styles.textColumtwoRowtwo}>Required</Text>
          </View>
          <View>
            <Text style={styles.textColumOneRowTwo}>{required} KSM</Text>
          </View>
        </View>
        <View style={styles.conversion}>
          <View>
            <Text style={styles.conversionText}>1 KSM = 5.80 USD</Text>
          </View>
        </View>
        <View style={styles.link}>
          <Link label="Cancel" color="#CC5741" />
          <Link
            label="Confirmed"
            color="#CC5741"
            action={() =>
              props.navigation.navigate('Confirmed', {
                type: props.route.params.typeOffer,
              })
            }
          />
        </View>
      </FooterWhite>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingVertical: '10%',
    paddingTop: '8%',
    paddingBottom: '33%',
  },
  containerColum: {
    flexDirection: 'column',
    borderTopRightRadius: 50,
    borderTopLeftRadius: 50,
    justifyContent: 'center',
    marginVertical: '5%',
    paddingTop: '5%',
    paddingBottom: '10%',
    backgroundColor: 'white',
    height: '43%',
    fontFamily: 'Poppins-Medium',
  },
  colum: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  textTitle: {
    color: 'white',
    fontSize: 20,
    fontFamily: 'Poppins-SemiBold',
    paddingTop: 20,
  },
  text: {
    color: 'white',
    fontSize: 16,
    fontFamily: 'Poppins-Medium',
    paddingTop: 10,
  },
  textColumOneRowOne: {
    paddingRight: '10%',
    fontSize: 14,
    fontWeight: 'bold',
    fontFamily: 'Poppins-Regular',
  },
  textColumtwoRowtwo: {
    paddingRight: '7%',
    fontWeight: 'bold',
    fontSize: 14,
    fontFamily: 'Poppins-Medium',
  },
  textColumOneRowTwo: {
    paddingLeft: '9%',
    fontSize: 14,
    fontFamily: 'Poppins-Medium',
  },
  conversion: {
    marginTop: '5%',
    paddingTop: '5%',
    marginBottom: '5%',
    paddingBottom: '5%',
    alignItems: 'center',
    borderTopWidth: 1,
    borderTopColor: '#C9C9C9',
    borderBottomWidth: 1,
    borderBottomColor: '#C9C9C9',
    width: '70%',
    marginLeft: '15%',
  },
  conversionText: {
    fontWeight: 'bold',
    fontSize: 14,
  },
  link: {
    alignItems: 'center',
  },
});

export default ConfirmationBuySell;
