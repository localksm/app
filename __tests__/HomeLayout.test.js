import 'react-native';
import React from 'react';
import { act } from 'react-test-renderer';
import { fireEvent, render } from 'react-native-testing-library';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { HomeLayout } from '../src/organisms';

configure({ adapter: new Adapter() });

it('layout renders correctly', async () => {
  jest.useFakeTimers();
  await act(() => {
    const wrapper = shallow(<HomeLayout />);
    expect(wrapper).toMatchSnapshot();
  });
});
