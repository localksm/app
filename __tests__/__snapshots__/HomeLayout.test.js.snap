// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout renders correctly 1`] = `ShallowWrapper {}`;
